src/functions/Encoding.ts

```diff
@@ -1,57 +1,71 @@
-import { BAD_PARAMS } from '../Oidc4vciErrors';
-import { IssuanceInitiationRequestPayload } from '../types';
+import { BAD_PARAMS, SearchValue } from '../types';
 
-export function encodeJsonAsURI(json: IssuanceInitiationRequestPayload[] | IssuanceInitiationRequestPayload) {
+/**
+ * @function encodeJsonAsURI encodes a Json object into a URI
+ * @param json object or array of type different objects.
+ */
+export function encodeJsonAsURI(json: unknown[] | unknown): string {
   if (!Array.isArray(json)) {
     return encodeJsonObjectAsURI(json);
   }
   return json.map((j) => encodeJsonObjectAsURI(j)).join('&');
 }
 
-function encodeJsonObjectAsURI(json: IssuanceInitiationRequestPayload) {
+function encodeJsonObjectAsURI(json: unknown): string {
   if (typeof json === 'string') {
     return encodeJsonObjectAsURI(JSON.parse(json));
   }
 
   const results = [];
 
-  function encodeAndStripWhitespace(key: string) {
+  function encodeAndStripWhitespace(key: string): string {
     return encodeURIComponent(key.replace(' ', ''));
   }
 
   for (const [key, value] of Object.entries(json)) {
     if (!value) {
       continue;
     }
+    //Skip properties that are not of URL type
+    if (!['issuer', 'credential_type'].includes(key)) {
```

Comment:
Should be rebased on @maikel-maas' PR, as this shouldn't present anymore

Reply:
reabsed again.